docs/analyticslib.md

```diff
@@ -2,39 +2,46 @@
 
 A set of analytical functions is provided for the simplification of common operations. This set of tools is targeted at new kdb+ developers or business users. Data visualization would be an example application of this library, as an effective visualization of data incurs various manipulations of data sets. For example, a user wishes to produce a 15 minute bucketed sample of a time series, and forward fill it, then pivot it. All of the above can be achieved with the functions defined below. 
 
-All functions contain usage examples, as well as an explanation on the type and form of the arguments they accept.
+## General Usage
+These set of function are defined in the `.al` namespace. All of the functions below have an input parameter in the form of a dictionary. The specifics of each dictionary are detailed in the *usage* section of each function. 
 
+This set of utilities also includes a general set of error traps which were designed to give informative errors specific to each function. For example if a dictionary has not been provided then the error would be:
 
-## ffills[]
+```
+q).al.pivot[123]
+'Input parameter must be a dictionary with keys:
+        -table
+        -by
+        -piv
+        -var
+```
+
+These error traps and messages are to guide the user in the correct use of the utilities and are designed to be as informative as possible.
+
+## ffill
 
 ### Usage
 This script contains the utility to dynamically forward fill a given table keyed by given columns.
 Input parameters:
-* Dictionary containing:
-	* `table` - Table to be forward filled
-	* `by` - list of column names to key the table (optional)
-	* `keycols` - list of columns names to be forward filled (optional)
+- Dictionary containing:
+	- table - Table to be forward filled
+	- by 	- list of column names to key the table (optional)
+	- keycols - list of columns names to be forward filled (optional)
```

Comment:
Do `by` and `table` need to be proceeded by extra spaces to align this correctly?